### PR TITLE
Document ortho_config v0.8.0 semantics and client-side domain validation (2.3.1)

### PR DESCRIPTION
**Pull request by leynos**
<pr-body>
## Summary
- Documents ortho_config v0.8.0 semantics and the final client-side domain validation contract. 
- Adds an ExecPlan doc for 2.3.1: Validate domains client-side before daemon startup. 
- Reworks preflight UX: now shows a canonical list of valid domains and a single optional did-you-mean suggestion; no operation catalogue is emitted; unknown-domain handling is implemented before config loading/daemon startup. 
- Adds canonical three-domain set: observe, act, verify; bounded Levenshtein-based distance threshold 2. 
- Localization/messages updated to support new guidance flow and wording. 
- Tests extended to cover unknown-domain handling, typo suggestions, and preflight guidance without daemon interaction. 
- Documentation updates across weaver-design.md, users-guide.md, and roadmap.md; ExecPlan doc added and maintained as a living document.

## Changes
- ExecPlan documentation:
  - Added docs/execplans/2-3-1-validate-domains-client-side-before-daemon-startup.md
- CLI and discoverability:
  - Introduces a canonical three-domain set: observe, act, verify
  - Adds a bounded Levenshtein-based suggestion helper with threshold 2
  - Reworks preflight guidance output to:
    - error: unknown domain '<domain>'
    - blank line
    - Valid domains: observe, act, verify
    - Optional Did you mean '<domain>'? (only if exactly one valid domain within distance 2)
    - No operation catalogue, no daemon-start messages
  - Updates handle_preflight and related flow to emit preflight guidance before config loading/daemon startup
- Refactors and error handling:
  - Renames MissingOperationGuidance to PreflightGuidance and updates related exit-code mapping
  - Updates Lib/Runner wiring to map PreflightGuidance as a failure exit when guidance is written
- Localization and messaging:
  - crates/weaver-cli/locales/en-US/messages.ftl updated with new IDs for valid domains and optional did-you-mean guidance
- Tests:
  - Adds unit tests for the bounded Levenshtein helper and domain suggestion logic
  - Updates unit tests to reflect new PreflightGuidance flow
  - Updates integration/behavior tests and main entry tests to verify unknown-domain handling and typo suggestions without daemon interaction
- Documentation and roadmap:
  - Updates docs/weaver-design.md and docs/users-guide.md to reflect final client-side validation behavior
  - Updates docs/roadmap.md to mark 2.3.1 as DONE
  - ExecPlan doc added and will be maintained as a living document for roadmap-driven changes
- Other:
  - Updates to main test entry points to assert the new guidance path and exit codes
  - Various test fixtures and feature files adjusted to reflect the new guidance contract

## Rationale
- Establishes a clear client-side contract: any invocation with an invalid first token (domain) must fail client-side before any configuration loading or daemon interaction, with a definitive list of valid domains and an optional single suggestion for close typos. This aligns CLI behavior with roadmap item 2.3.1 and reduces exposure to daemon startup until the contract is satisfied.

## Content overview of the ExecPlan (highlights)
- Purpose: enforce a strict client-side contract for unknown-domain handling on the CLI prior to any daemon interaction.
- Constraints, Tolerances, Risks: enumerates non-breaking changes, failure modes, and escalation criteria.
- Plan of work: Stage A through Stage D with gating criteria and testing scope.
- Context and orientation: maps the CLI surface and relevant components for domain validation.
- References: follow-up work in design/docs updates.
- Roadmap reference: ExecPlan: 2-3-1 Validate domains client-side before daemon startup; Roadmap item 2.3.1: DONE.

## Plan of work (high level)
- Stage A: lock the contract in tests first (red phase) by adding/adjusting unit, integration, and BDD tests before code changes.
  - Stage A tests cover: unknown-domain, unknown-domain with operation, close typo with a single suggestion, distant typo without suggestion, edge cases for distance/suggestion helper, and ensure daemon interaction is not leaked in preflight paths.
- Stage B: implement shared preflight validation and suggestion logic (green phase).
  - Introduces helper in crates/weaver-cli/src/discoverability.rs to normalize tokens, recognize domains, compute distance within threshold, and return renderable results.
  - Broadens preflight predicate to cover known-domain-without-operation and unknown-domain paths.
  - Reworks output to show valid domains, optional suggestion, and no operation catalogue.
  - Rewrites handle_preflight flow to emit guidance prior to config loading/daemon startup.
  - Renames sentinel to PreflightGuidance and update tests
  - Update Lib/Runner etc.
- Documentation updates: ... etc.

### Stage C: document the contract and mark the roadmap item complete
+
- Update the design and operator docs once the code and tests are stable.
+
- Stage D: run the full gate sequence and capture evidence
+
- Go/no-go gates described in ExecPlan.

## References
- ExecPlan: 2-3-1 Validate domains client-side before daemon startup
- Roadmap item 2.3.1: DONE

📎 **Task**: https://www.devboxer.com/task/662331d0-4f89-48f5-9a8b-3d2c729d1fde
</pr-body>

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[crates/weaver-cli/src/tests/unit/missing_operation_guidance.rs](https://github.com/leynos/weaver/pull/88/files/5a3c2ea265bc1edc39d78f0f7b87b8d4ccd56e4c#diff-a6a28940c9a9e05ca4c47c8d06e3c4b253a92cf2514841b64281073cf5f96699)

Comment on file

~~~rust
    assert!(stderr_text.contains("command domain"));
    assert!(!stderr_text.contains("error: operation required for domain"));
    assert!(!stderr_text.contains("Available operations:"));
    assert!(!stderr_text.contains("Valid domains: observe, act, verify"));
~~~

❌ New issue: [Duplicated Assertion Blocks](https://codescene.io/projects/71837/delta?repo-id=1205553&review-id=88&biomarker=Duplicated+Assertion+Blocks&filename=crates%2Fweaver-cli%2Fsrc%2Ftests%2Funit%2Fmissing_operation_guidance.rs&method=)
The test suite contains 4 functions with duplicated assertion blocks (distant_unknown_domain_omits_suggestion,typo_domain_emits_single_suggestion,unknown_domain_with_operation_emits_global_guidance_before_configuration_loading,unknown_domain_without_operation_emits_global_guidance), threshold = 2

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[crates/weaver-cli/src/tests/unit/missing_operation_guidance.rs](https://github.com/leynos/weaver/pull/88/files/5a3c2ea265bc1edc39d78f0f7b87b8d4ccd56e4c#diff-a6a28940c9a9e05ca4c47c8d06e3c4b253a92cf2514841b64281073cf5f96699)

Comment on file

~~~rust
    assert!(stderr_text.contains("command domain"));
    assert!(!stderr_text.contains("error: operation required for domain"));
    assert!(!stderr_text.contains("Available operations:"));
    assert!(!stderr_text.contains("Valid domains: observe, act, verify"));
~~~

❌ New issue: [Large Assertion Blocks](https://codescene.io/projects/71837/delta?repo-id=1205553&review-id=88&biomarker=Large+Assertion+Blocks&filename=crates%2Fweaver-cli%2Fsrc%2Ftests%2Funit%2Fmissing_operation_guidance.rs&method=)
The test suite contains 6 assertion blocks with at least 4 assertions, threshold = 4

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[crates/weaver-cli/src/tests/unit/missing_operation_guidance.rs](https://github.com/leynos/weaver/pull/88/files/5a3c2ea265bc1edc39d78f0f7b87b8d4ccd56e4c#diff-a6a28940c9a9e05ca4c47c8d06e3c4b253a92cf2514841b64281073cf5f96699)

Comment on file

~~~rust
    assert!(stderr_text.contains("Valid domains: observe, act, verify"));
    assert!(!stderr_text.contains("Available operations:"));
    assert!(!stderr_text.contains("weaver observe get-definition --help"));
}
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/71837/delta?repo-id=1205553&review-id=88&biomarker=Code+Duplication&filename=crates%2Fweaver-cli%2Fsrc%2Ftests%2Funit%2Fmissing_operation_guidance.rs&method=)
The module contains 5 functions with similar structure: distant_unknown_domain_omits_suggestion,known_domain_without_operation_emits_contextual_guidance,typo_domain_emits_single_suggestion,unknown_domain_with_operation_emits_global_guidance_before_configuration_loading and 1 more functions

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[crates/weaver-cli/src/lib.rs](https://github.com/leynos/weaver/pull/88/files/5a3c2ea265bc1edc39d78f0f7b87b8d4ccd56e4c#diff-17a30b0584d2348541b6621fe97320b4f869125a087efff03bf2775b20245ba9)

Comment on file

~~~rust
        match KnownDomain::try_parse(raw_domain) {
            Some(domain)
                if cli
                    .operation
                    .as_deref()
                    .is_none_or(|operation| operation.trim().is_empty()) =>
            {
                if write_missing_operation_guidance(stderr, localizer, domain)
                    .map_err(AppError::EmitGuidance)?
                {
                    return Err(AppError::PreflightGuidance);
                }
            }
            Some(_) => {}
            None => {
                if write_unknown_domain_guidance(stderr, localizer, raw_domain)
                    .map_err(AppError::EmitGuidance)?
                {
                    return Err(AppError::PreflightGuidance);
                }
            }
~~~

❌ New issue: [Deep, Nested Complexity](https://codescene.io/projects/71837/delta?repo-id=1205553&review-id=88&biomarker=Deep%2C+Nested+Complexity&filename=crates%2Fweaver-cli%2Fsrc%2Flib.rs&method=handle_preflight)
handle_preflight has a nested complexity depth of 4, threshold = 4

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[crates/weaver-cli/src/tests/unit/missing_operation_guidance.rs](https://github.com/leynos/weaver/pull/88/files/cccfdda515ef7b8afcb01af27ccc62127f1afcc9#diff-a6a28940c9a9e05ca4c47c8d06e3c4b253a92cf2514841b64281073cf5f96699)

Comment on file

~~~rust

#[test]
fn known_domain_without_operation_emits_contextual_guidance() {
fn run_with_panicking_loader(args: &[&str]) -> (ExitCode, Vec<u8>, String) {
~~~

❌ New issue: [String Heavy Function Arguments](https://codescene.io/projects/71837/delta?repo-id=1205553&review-id=88&biomarker=String+Heavy+Function+Arguments&filename=crates%2Fweaver-cli%2Fsrc%2Ftests%2Funit%2Fmissing_operation_guidance.rs&method=)
In this module, 50.0% of all arguments to its 11 functions are strings. The threshold for string arguments is 39.0%

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[crates/weaver-cli/src/tests/unit/missing_operation_guidance.rs](https://github.com/leynos/weaver/pull/88/files/cccfdda515ef7b8afcb01af27ccc62127f1afcc9#diff-a6a28940c9a9e05ca4c47c8d06e3c4b253a92cf2514841b64281073cf5f96699)

Comment on file

~~~rust
fn assert_unknown_domain_preflight(exit: ExitCode, stdout: &[u8], stderr_text: &str, domain: &str) {
    assert_eq!(exit, ExitCode::FAILURE);
    assert!(stdout.is_empty(), "guidance must not write to stdout");
    assert!(stderr_text.contains(&format!("error: unknown domain '{domain}'")));
    assert!(stderr_text.contains("Valid domains: observe, act, verify"));
}
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/71837/delta?repo-id=1205553&review-id=88&biomarker=Code+Duplication&filename=crates%2Fweaver-cli%2Fsrc%2Ftests%2Funit%2Fmissing_operation_guidance.rs&method=)
The module contains 2 functions with similar structure: assert_known_domain_operation_guidance,assert_unknown_domain_preflight

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

## String Heavy Function Arguments
crates/weaver-cli/src/tests/unit/missing_operation_guidance.rs: 

### What lead to degradation?
In this module, 50.0% of all arguments to its 11 functions are strings. The threshold for string arguments is 39.0%

### Why does this problem occur?
String is a generic type that fail to capture the constraints of the domain object it represents. In this module, 50 % of all function arguments are string types.

### How to fix it?
Heavy string usage indicates a missing domain language. Introduce data types that encapsulate the semantics. For example, a user_name is better represented as a constrained User type rather than a pure string, which could be anything.

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

## Code Duplication
crates/weaver-cli/src/tests/unit/missing_operation_guidance.rs: 

### What lead to degradation?
The module contains 2 functions with similar structure: assert_known_domain_operation_guidance,assert_unknown_domain_preflight

### Why does this problem occur?
Duplicated code often leads to code that's harder to change since the same logical change has to be done in multiple functions. More duplication gives lower code health.

### How to fix it?
A certain degree of duplicated code might be acceptable. The problems start when it is the same behavior that is duplicated across the functions in the module, ie. a violation of the Don't Repeat Yourself (DRY) principle. DRY violations lead to code that is changed together in predictable patterns, which is both expensive and risky. DRY violations can be identified using CodeScene's X-Ray analysis to detect clusters of change coupled functions with high code similarity. [Read More](https://codescene.com/blog/software-revolution-part3/)
Once you have identified the similarities across functions, look to extract and encapsulate the concept that varies into its own function(s). These shared abstractions can then be re-used, which minimizes the amount of duplication and simplifies change.

File: crates/weaver-cli/src/discoverability.rs [hunk 1]
```diff
@@ -42,6 +42,12 @@ impl KnownDomain {
             .map(|(_, _, ops)| *ops)
             .unwrap_or_else(|| panic!("missing DOMAIN_OPERATIONS entry for '{}'", self.as_str()))
     }
+
+    fn catalogue_order() -> impl Iterator<Item = Self> {
+        DOMAIN_OPERATIONS
+            .iter()
+            .map(|(domain, _, _)| known_domain_from_catalogue_entry(domain))
+    }
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/weaver-cli/src/discoverability.rs:55]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: docs/weaver-design.md [hunk 2]
```diff
@@ -417,9 +417,16 @@ Known-domain invocations that omit the operation (for example,
...
+startup, or transport setup.
+
+Unknown domains follow a separate client-side preflight rule. Any invocation
+whose first positional token is not one of the canonical domains `observe`,
+`act`, or `verify` fails before configuration discovery, daemon startup, or
+transport setup regardless of whether an operation token is present. The error
+body always includes `Valid domains: observe, act, verify`. When exactly one
+valid domain is within edit distance 2 of the supplied token, the CLI appends a
+single deterministic `Did you mean ...?` suggestion; otherwise it emits no
+suggestion.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/weaver-design.md:429]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: crates/weaver-cli/src/tests/unit/missing_operation_guidance.rs [hunk 3]
```diff
@@ -18,50 +18,125 @@ impl ConfigLoader for PanickingLoader {
...
+    assert!(output.stderr.contains("Did you mean 'observe'?"));
+}
+
+#[test]
+fn distant_unknown_domain_omits_suggestion() {
+    let output = run_with_panicking_loader(&["bogus", "get-definition"]);
+
+    assert_unknown_domain_preflight(&output, "bogus");
+    assert!(!output.stderr.contains("Did you mean"));
 }
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/weaver-cli/src/tests/unit/missing_operation_guidance.rs:163]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: docs/users-guide.md [hunk 4]
```diff
@@ -66,12 +66,13 @@ directive = "force"
...
-with a `LoadConfiguration` error that lists every offending path. Remove or fix
-the reported files before retrying. If no configuration files exist at all the
-loader still falls back to the built-in defaults described below.
+`weaver` now uses `ortho_config` v0.8.0 and Rust 1.88. The `ortho_config` v0.8.0
+loader treats invalid configuration files as fatal. When `--config-path` points
+at a broken file, or when discovery finds a malformed `weaver.toml`/`.weaver.toml`,
+both the CLI and daemon abort with a `LoadConfiguration` error that lists every
+offending path. Remove or fix the reported files before retrying. If no
+configuration files exist at all the loader still falls back to the built-in
+defaults described below.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/users-guide.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: docs/weaver-design.md [hunk 5]
```diff
@@ -1005,17 +1012,17 @@ Configuration is layered with `ortho-config`, producing the precedence order
...
+v0.8.0 switch lets `weaver_config::Config` declare its discovery policy inline
+through the `#[ortho_config(discovery(...))]` attribute. The app name, dotfile,
+project file, and `--config-path` flag are all defined next to the struct, so
+every consumer shares the same generated loader without bespoke builders.
+
+The `ortho_config` v0.8.0 loader preserves the stricter discovery and parsing
+model adopted in earlier releases: if any discovered configuration file fails to
+parse, `ConfigDiscovery::load_first` returns an aggregated `OrthoError`. Both
+the CLI and daemon bubble that error to the user instead of quietly falling back
+to defaults, making misconfigurations immediately visible.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/weaver-design.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: docs/users-guide.md [hunk 6]
```diff
@@ -66,12 +66,13 @@ directive = "force"
...
-with a `LoadConfiguration` error that lists every offending path. Remove or fix
-the reported files before retrying. If no configuration files exist at all the
-loader still falls back to the built-in defaults described below.
+`weaver` now uses `ortho_config` v0.8.0 and Rust 1.88. The `ortho_config` v0.8.0
+loader treats invalid configuration files as fatal. When `--config-path` points
+at a broken file, or when discovery finds a malformed `weaver.toml`/`.weaver.toml`,
+both the CLI and daemon abort with a `LoadConfiguration` error that lists every
+offending path. Remove or fix the reported files before retrying. If no
+configuration files exist at all, the loader still falls back to the built-in
+defaults described below.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/users-guide.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

[the remark this answers is not in the document]
Comment (leynos) [reply docs/users-guide.md]: @coderabbitai please check `docs/developers-guide.md`
